source/dagger/gameplay/ping_pong/pingpong_ball.cpp

```diff
@@ -60,7 +60,14 @@ void PingPongBallSystem::Run()
                 if (std::abs(collisionSides.y) > 0)
                 {
                     ball.speed.y *= -1;
-                }
+                }  
+            }
+
+            if (ball.isMalicious == true &&  Engine::Registry().has<ControllerMapping>(col.colidedWith)  ) {
+               auto &cntrl =  Engine::Registry().get<ControllerMapping>(col.colidedWith);
+               cntrl.inverted = -1;
+               Engine::Registry().destroy(entity);
```

Comment:
@gavrilovmiroslav , what do u think, should we do that in EndOfFrame or it's fine here?

Reply:
I had a case where it actually made a difference, so I'd play it safe and delegate to EndOfFrame. Maybe we can even make a small helper mechanism for this behavior sometime later so that we don't have to always do this ourselves...

Reply:
@Coakim , put some flag into ball instead of deleting it and then in EndOfFrame function delete it.

You can find an example how to subscribe to EndFrame event here:
RacingCollisionsLogicSystem::OnEndOfFrame

Reply:
I was just looking for a way to destroy a specific ball and found the Engine::Registry().destroy function , but I guess I didn't think about whether it could cause problems doing it mid frame. Will move the deletion to EndOfFrame function as suggested

Reply:
Moved the deletion to EndOfFrame